x-pack/plugins/osquery/public/plugin.ts

```diff
@@ -32,38 +32,58 @@ import {
   LazyOsqueryManagedPolicyEditExtension,
   LazyOsqueryManagedCustomButtonExtension,
 } from './fleet_integration';
+import type { SecurityPluginStart } from '../../security/public';
 
 export function toggleOsqueryPlugin(
   updater$: Subject<AppUpdater>,
   http: CoreStart['http'],
-  registerExtension?: StartPlugins['fleet']['registerExtension']
+  registerExtension?: StartPlugins['fleet']['registerExtension'],
+  authc?: SecurityPluginStart['authc']
 ) {
-  http
-    .fetch<GetPackagesResponse>(epmRouteService.getListPath(), { query: { experimental: true } })
-    .then(({ response }) => {
-      const installed = response.find(
-        (integration) =>
-          integration?.name === OSQUERY_INTEGRATION_NAME && integration?.status === 'installed'
-      );
-
-      if (installed && registerExtension) {
-        registerExtension({
-          package: OSQUERY_INTEGRATION_NAME,
-          view: 'package-detail-custom',
-          component: LazyOsqueryManagedCustomButtonExtension,
-        });
-      }
-
-      updater$.next(() => ({
-        navLinkStatus: installed ? AppNavLinkStatus.visible : AppNavLinkStatus.hidden,
-      }));
-    })
-    .catch(() => {
+  const toggleOsqueryPluginAsync = async () => {
+    let currentUser;
+    try {
+      currentUser = await authc?.getCurrentUser();
```

Comment:
This is unfortunately trading one problem for another. We also don't want to retrieve the current user on authenticated pages -- we actually just resolved https://github.com/elastic/kibana/issues/99503 late last week, and this would end up re-introducing that problem.

What we need to do instead, is use Core's `anonymous` service to detect if the browser is currently on an anonymous path. If so, we will skip initialization.

Here are a couple of examples of how the `anonymous` service is used elsewhere to prevent these requests from firing on the login page (and other anonymous pages):

https://github.com/elastic/kibana/blob/12bb59abb030f2c8e7ffa3464e85f00429212bf1/x-pack/plugins/saved_objects_tagging/public/plugin.ts#L69-L95

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/x-pack/plugins/licensing/public/services/feature_usage_service.ts#L53-L58

https://github.com/elastic/kibana/blob/0886b0e9e5c970233ad18dccd8c96fded2198b1e/x-pack/plugins/cloud/public/plugin.ts#L121-L137

Reply:
Thank you @legrego! I hope it looks better now :)